Document the Task Updates Endpoint

JeffBain opened the pull request:
Added Docs for the Task Updates Endpoint. Feel free to nitpick some of the language or clarify things.

funkaoshi commented:
Should tidy up the formatting to match the latest stuff @sibartlett pushed up.

funkaoshi commented:
@ianpersaud your stuff should be here as well. The goal was to document first, and then code.

funkaoshi commented:
Can you fix the formatting to match what Simon has been doing. More blank lines between sections, etc.

funkaoshi commented:
This seems fine otherwise.

geoffwhittington commented on source/includes/task_updates.md:
```diff
@@ -0,0 +1,162 @@
+# Task Updates
+
+
+## Get all unaccepted changes to the tasks for a project
```
How does this sound? "Retrieve the list of tasks relevant to a project but not yet accepted for inclusion by a project user"

geoffwhittington replied:
Maybe I don't understand this correctly...

funkaoshi replied:
That's correct, and better wording.

JeffBain replied:
Apparently wasn't getting emails for this thread, but I'm going through this soon to merge the docs changes. Only issue with this is that it's both types of changes, so relevant but unaccepted tasks and tasks that used to be relevant but are no longer. Any preferred way to describe those as well?

geoffwhittington commented on source/includes/task_updates.md:
```diff
@@ -0,0 +1,162 @@
+relevant to a project due to changes in the rules/active status, and tasks that
+have become relevant since the last time the survey was updated for similar reasons.
+
+**`GET /api/v2/projects/{project_id}/tasks-updates/`**
```
I'm sure people are eventually going to want to only accept updates for certain tasks - not the whole lot. Could this endpoint be updated in the future to accept a list of task_id's?

geoffwhittington commented:
I assume the plan for the `description` attribute is to inline the task amendmends?

JeffBain commented:
I updated the docs now. I tried to capture the language Geoff used, though I used that more in the description than the header, as I needed something more general for the header + it would have gotten too long to describe both types of changes. 

I looked at one of Simon's commits to get an idea what his formatting was like, so I think I did some of that.

I dropped the endpoint documentation about changing individual task's acceptance status. I still want to do that, but while writing that we ran into the issue that there's no way for us to queue up and combine task update emails, so it would have created an email per acceptance. I think we need to switch that to something that works a little bit more smartly before we give users the option.

funkaoshi commented on source/includes/task_updates.md:
````diff
@@ -0,0 +1,113 @@
+
+## Retrieve the list of tasks whose inclusion state does not match its relevance
+
+```http
````
You can put the ``` on their own line.

funkaoshi commented on source/includes/task_updates.md:
```diff
@@ -0,0 +1,113 @@
+Parameter  | Description
+---------- | ------------
+project_id | The ID of the project for the tasks
+
```
Simon has big gaps between each command.

funkaoshi commented on source/includes/task_updates.md:
```diff
@@ -0,0 +1,113 @@
+GET /api/v2/projects/1/task-updates/1-T2/ HTTP/1.1
+Accept: application/json
+Authorization: Token "YOUR SDE ACCESS TOKEN"
+
```
Superfluous blank lines should be removed.

JeffBain replied:
I swear in the Simon commit I looked like he added lines like these.

We really need an actual style guide for these docs as opposed to the tribal knowledge thing.

sibartlett replied:
There shouldn't be blank lines at the end of example blocks - if it was like that in my docs, it was a mistake.

Yesterday I started writing a node script that fixes and standardizes blank lines in our docs.

funkaoshi commented:
@JeffBain i'm going to merge this, because Vey is waiting on changes, but can you update the docs so the markdown is cleaned up.
